src/zjs_callbacks.c

```diff
@@ -315,6 +334,20 @@ void zjs_remove_callback(int32_t id)
     }
 }
 
+void zjs_signal_callback_args(int32_t id, void* args, uint32_t size)
+{
+    if (args && size) {
+        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)size / 4);
```

Comment:
I think we discussed this on IRC but I will reiterate it here. The only problem with using ONLY the ring buffer as the pending callback queue is that you have the know the size of the ring buffer element before hand, which is why I use the queue to hold that size. I agree that this is not correct because you could push two of the same callback ID's with different argument sizes. We essentially need a Zephyr API that lets you get the size of the next ring buffer element, then we could create an array that is big enough to hold the argument data. You had said that we could just have a set size (like 4 or 5) but this seem like more of a hack and not very efficient. I think it would be quite simple to add a Zephyr API which lets you get the size of the next ring buffer element (or we could just hack it and look into the ring buffer's data and find the size.